core/profiles/src/main/java/software/amazon/awssdk/profiles/ProfileFileSupplier.java

```diff
@@ -0,0 +1,160 @@
+     * disk file is unchanged will return the same object.
+     *
+     * @param path Path to the file to read from.
+     * @param type The type of file. See {@link ProfileFile.Type} for possible values.
+     * @return Implementation of {@link ProfileFileSupplier} that is capable of supplying a new profile when the file
+     *         has been modified.
+     */
+    static ProfileFileSupplier reloadWhenModified(Path path, ProfileFile.Type type) {
+        return new ProfileFileSupplier() {
+
+            final ProfileFile.Builder builder = ProfileFile.builder()
+                                                           .content(path)
+                                                           .type(type);
+
+            final ProfileFileRefresher refresher = ProfileFileRefresher.builder()
+                                                                       .profileFile(builder::build)
+                                                                       .profileFilePath(path)
+                                                                       .build();
+
+            @Override
+            public ProfileFile get() {
+                return refresher.refreshIfStale();
+            }
+
+            @Override
+            public void close() {
+                refresher.close();
+            }
+        };
+    }
+
+    /**
+     * Creates a {@link ProfileFileSupplier} that produces an existing profile.
+     *
+     * @param profileFile Profile object to supply.
+     * @return Implementation of {@link ProfileFileSupplier} that is capable of supplying a single profile.
+     */
+    static ProfileFileSupplier fixedProfileFile(ProfileFile profileFile) {
+        return () -> profileFile;
+    }
+
+    /**
+     * Creates a {@link ProfileFileSupplier} by combining the {@link ProfileFile} objects from two {@code ProfileFileSupplier}s.
+     * Objects are passed into {@link ProfileFile.Aggregator}.
+     *
+     * @param suppliers Array of {@code ProfileFileSupplier} objects. {@code ProfileFile} objects are passed to
+     *                  {@link ProfileFile.Aggregator#addFile(ProfileFile)} in the same argument order as the supplier that
+     *                  generated it.
+     * @return Implementation of {@link ProfileFileSupplier} aggregating results from the supplier objects.
+     */
+    static ProfileFileSupplier aggregate(ProfileFileSupplier... suppliers) {
+
+        return new ProfileFileSupplier() {
+
+            final AtomicReference<ProfileFile> currentAggregateProfileFile = new AtomicReference<>();
+
+            @Override
+            public ProfileFile get() {
+                ProfileFile.Aggregator aggregator = ProfileFile.aggregator();
+                for (ProfileFileSupplier supplier : suppliers) {
+                    aggregator.addFile(supplier.get());
+                }
+
+                return refreshAndGetCurrentAggregate(aggregator);
```

Comment:
Do we really need to re-aggregate with every profile file get()?

Reply:
Got it, yeah if we can avoid re-aggregating when the files haven't changed that's a good fix.

Reply:
Reimplemented previous code that was looking at whether the individual files had changes. With this new version, we should not re-aggregate if not needed.